code/modules/clothing/suits/armor.dm

```diff
@@ -387,10 +390,12 @@
 		E.Goto(owner, E.move_to_delay, E.minimum_distance)
 		owner.alpha = 0
 		owner.visible_message("<span class='danger'>[owner] is hit by [attack_text] in the chest!</span>") //We pretend to be hit, since blocking it would stop the message otherwise
-		spawn(40)
-			owner.alpha = initial(owner.alpha)
+		addtimer(CALLBACK(src, .proc/remove_invisible, owner), 4 SECONDS)
 		return TRUE
 
+/obj/item/clothing/suit/armor/reactive/stealth/proc/remove_invisible(mob/living/carbon/human/owner)
+	owner.alpha = initial(owner.alpha)
```

Comment:
How does this interact with a vampire that moves into darkness with shadow cloak on during the stealth period?

Reply:
it resets it for a brief moment, but then it goes back to being cloaked.

Reply:
Consider this resolved